internal/keepers/service.go

```diff
@@ -9,36 +9,51 @@ import (
 	"sync"
 	"time"
 
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+
```

Comment:
I would like to keep any chain specific logic out of the plugin. Capture chain specific logic in the pkg/chain package and only consume common types in the plugin directly.